### ISSUE TITLE
[openSUSE Tumbleweed] error: no matching function for call to 'JSONItem::JSONItem(const wxString&, wxString, int)'

### ISSUE THREAD

**Issue by kevinsmia1939** [eranif/codelite#2617]
Hello,

I am trying to update codelite 12.0.0 to 14.0.2 on openSUSE Tumbleweed.

I encounter 2 errors.
```

[  320s] /home/abuild/rpmbuild/BUILD/codelite-14.0.2/CodeLite/JSON.cpp:435:56: error: no matching function for call to 'JSONItem::JSONItem(const wxString&, wxString, int)'
[  320s]   435 |     append(JSONItem(name, wxString(value), cJSON_String));
[  320s]       |                                                        ^

[  320s] /home/abuild/rpmbuild/BUILD/codelite-14.0.2/CodeLite/JSON.cpp:455:102: error: no matching function for call to 'JSONItem::JSONItem(const wchar_t [1], const wxString&, int)'
[  320s]   455 | void JSONItem::arrayAppend(const wxString& value) { arrayAppend(JSONItem(wxT(""), value, cJSON_String)); }
[  320s]       |                                                                                                      ^

```
Full log here: https://build.opensuse.org/build/home:andythe_great:branches:devel:tools/openSUSE_Tumbleweed/x86_64/codelite/_log

Dependencies that I think is important.
```
gcc-c++-10-3.1 
clang-devel-11.0.0-1.1
wxWidgets version is: 3.0.4
gtk3-devel-3.24.23-1.1
gtk2-devel-2.24
```

Thank you.

**Comment by dghart** [comment]
Hi,

>I am trying to update codelite 12.0.0 to 14.0.2

While 14.0.2 is the latest git tag,  the latest 'weekly' build is 14.0.4.

> Full log here: https://build.opensuse.org/build/home:andythe_great:branches:devel:tools/openSUSE_Tumbleweed/x86_64/codelite/_log

That page is only available if you are logged in...

I don't have tumbleweed here but, on an openSUSE 15.2 virtuabox guest, I just successfully built CodeLite git head against a gtk+3 wx3.0.4 build. (BTW, the latest 'stable' wx release is 3.0.5.)

Maybe the bug has been fixed in the 5 months since the 14.0.2 tag. You can easily download a zip of head from github. If that also fails, please give more details of your CodeLite configuration settings and the wx3.0.4 configuration, and attach the CodeLite build log here.

Regards,

David

**Comment by kevinsmia1939** [comment]
> Hi,
> 
> > I am trying to update codelite 12.0.0 to 14.0.2
> 
> While 14.0.2 is the latest git tag, the latest 'weekly' build is 14.0.4.
> 
> > Full log here: https://build.opensuse.org/build/home:andythe_great:branches:devel:tools/openSUSE_Tumbleweed/x86_64/codelite/_log
> 
> That page is only available if you are logged in...
> 
> I don't have tumbleweed here but, on an openSUSE 15.2 virtuabox guest, I just successfully built CodeLite git head against a gtk+3 wx3.0.4 build. (BTW, the latest 'stable' wx release is 3.0.5.)
> 
> Maybe the bug has been fixed in the 5 months since the 14.0.2 tag. You can easily download a zip of head from github. If that also fails, please give more details of your CodeLite configuration settings and the wx3.0.4 configuration, and attach the CodeLite build log here.
> 
> Regards,
> 
> David

I will look into the issue first, might be issue with my build instruction, close the issue for now.
